visualization/include/visualization/svg_render.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <fstream>
+#include <iterator>
+#include <sstream>    
+
+#include "element.hpp"
+
+namespace visualization::render {
+
+    class SVGDimension {
+    public:
+        SVGDimension(double width, double height);
+
+        double get_width() const;
+        double get_height() const;
+
+    private: 
+        double width;
+        double height;
+    };
+
+    class SVGRender {
+    public:
+        virtual ~SVGRender();
+
+        SVGRender(const std::string name, const SVGDimension& dimension, const std::vector<builder::Element> elements);
+
+        const void render() const;
```

Comment:
`render` for me must return something and not save it into file because first, it's not the problem of SVGRenderer to save the svg to a file and second, it makes your class unecessarely untestable. Better to have a render method returning a string and the caller of the function will know better what to do with the output and if it needs to be saved or not.